image/manifest.go

```diff
@@ -240,3 +240,33 @@ loop:
 	}
 	return nil
 }
+
+// ValidateManifestMediaType validate the manifest schema media-type
+// and its descendants fields media type, such as config and layers,
+// to make sure they match to spec definition, or returns an error if
+// the validation failed.
+func ValidateManifestMediaType(r io.Reader) error {
```

Comment:
I don't see a reason to expose media-type validation as a separate thing.  There's just “validation”, which includes checking media types against spec requirements as well as checking other parameter values against spec requirements.  That probably means we want a comparison with the appropriate JSON Schema (to catch things like `schemaVersion` is unset) and additional checks for cases not covered by JSON Schema (e.g. `schemaVersion != 2`).  But I don't think a caller will care about picking and choosing particular subsets of validation.